src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,36 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            if (workspace is VisualStudioWorkspaceImpl vsWorkspace)
+            {
+                var results = ThreadingContext.JoinableTaskFactory.Run(async () =>
+                {
+                    return await spanMappingService.MapSpansAsync(generatedDocument, SpecializedCollections.SingletonEnumerable(textSpan), CancellationToken.None).ConfigureAwait(true);
```

Comment:
seeing all the UI work and the sync blocking on default cancellation tokens makes me sad.  can you open issue for us to make the navigation service cancellable.  Features that use it should pass their cancellation tokens in.  this way if this step is taking a long time, they can cancel and get back to whatever they were doing.

Reply:
Yup, filed https://github.com/dotnet/roslyn/issues/45971